ambari-server/src/main/resources/custom_actions/scripts/mpack_packages.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+"""
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+"""
+import signal
+
+import ambari_simplejson as json
+
+from ambari_commons.os_check import OSCheck
+from ambari_commons.shell import RepoCallContext
+from ambari_commons.str_utils import cbool, cint
+from ambari_commons.repo_manager import ManagerFactory
+from resource_management.core.exceptions import Fail
+from resource_management.core.logger import Logger
+from resource_management.libraries.functions.repository_util import CommandRepository
+from resource_management.libraries.script.script import Script
+
+
+class MpackPackages(Script):
+  """
+  This script is a part of Upgrade workflow
+  """
+
+  def __init__(self):
+    super(MpackPackages, self).__init__()
+
+    self.repo_mgr = ManagerFactory.get()
+    self.repo_files = {}
+
+  def actionexecute(self, env):
+    num_errors = 0
+
+    # Parse parameters
+    config = Script.get_config()
+
+    try:
+      command_repository = CommandRepository(config['repositoryFile'])
+    except KeyError:
+      raise Fail("The command repository indicated by 'repositoryFile' was not found")
+
+    # Handle a SIGTERM and SIGINT gracefully
+    signal.signal(signal.SIGTERM, self.abort_handler)
+    signal.signal(signal.SIGINT, self.abort_handler)
+
+    self.repository_version = command_repository.version_string
+
+    # Select dict that contains parameters
+    try:
+      package_list = json.loads(config['roleParams']['package_list'])
+      stack_id = config['roleParams']['stack_id']
+    except KeyError:
+      pass
+
+    self.stack_name = Script.get_stack_name()
+    if self.stack_name is None:
+      raise Fail("Cannot determine the stack name")
+
+    self.stack_root_folder = Script.get_stack_root()
+    if self.stack_root_folder is None:
+      raise Fail("Cannot determine the stack's root directory")
+
+    if self.repository_version is None:
+      raise Fail("Cannot determine the repository version to install")
+
```

Comment:
I don't think we need stack_name, stack_version and repo_version anymore, right? The commandRepository should be enough.

Reply:
Nope - this was leftover from my copy from `install_packages`.

Reply:
Done